tools/checklocks/analysis.go

```diff
@@ -548,23 +534,51 @@ func (pc *passContext) checkBasicBlock(fn *ssa.Function, block *ssa.BasicBlock,
 		}
 	}
 
-	// Analyze all successors.
-	for _, succ := range block.Succs {
-		// Collect possible return values, and make sure that the lock
-		// state aligns with any return value that we may have found
-		// above. Note that checkBasicBlock will recursively analyze
-		// the lock state to ensure that Releases and Acquires are
-		// respected.
-		if pls := pc.checkBasicBlock(fn, succ, lff, ls, seen); pls != nil {
-			if rls != nil && !rls.isCompatible(pls) {
+	returns = rls != nil
```

Comment:
I don't think you need rls really:
```
returns = false
if len(block.Instrs) > 0 {
  _, returns = block.Instrs[len(block.Instrs)-1].(*Return)
}
```
Even better is a function `Returns(bb *BasicBlock) bool`.

Reply:
(This kinda suggests returning post instead of rls for checkInstruction. But consider this more of a TODO.)

Reply:
Yeah, in the meantime, I eliminated the named returns in `checkBasicBlock` and implemented a function to check if a basic block returns.

Reply:
I am still seeing a returns bool. Am I looking at an old version?